packages/server-core/src/bot/bot/bot.hooks.ts

```diff
@@ -39,7 +44,29 @@ import {
   botResolver
 } from '../../bot/bot/bot.resolvers'
 import authenticate from '../../hooks/authenticate'
+import persistData from '../../hooks/persist-data'
 import verifyScope from '../../hooks/verify-scope'
+import { BotService } from './bot.class'
+
+async function sanitizeInstanceId(context: HookContext<BotService>) {
```

Comment:
This should not be in hook rather it should be done in data resolvers. Have a look of `userDataResolver` in `packages/server-core/src/user/user/user.resolvers.ts`